view/adminhtml/web/js/breadcheckout.js

```diff
@@ -37,20 +37,38 @@ define(
                                         timeout: false
                                     });
                                 }
-                            }).done(function (response) {
-                                try {
-                                    if (typeof response.result !== 'undefined' && response.result === true) {
-                                        $('#bread-checkout-btn').hide();
-                                        var approved = "<p><strong>You have been approved for financing.<br/>"+
-                                            "Please continue with the checkout to complete your order.</strong></p>";
-                                        $('#bread_feedback').html(approved);
-                                        $('body').trigger('hideLoadingPopup');
-                                    }
-                                } catch (e) {
-                                    console.log(e);
+                            }).done(function(response) {
+                                if (response.error) {
+                                    errorInfo = {
+                                        bread_config: breadConfig,
+                                        response: response,
+                                        tx_id: tx_token,
+                                    };
+                                    document.logIssue('error', errorInfo, 'Error validating payment method');
+
+                                    alert(response.error);
+                                } else if (response.result && response.result === true) {
+                                    $('#bread-checkout-btn').hide();
+                                    var approved = "<p><strong>You have been approved for financing.<br/>"+
+                                        "Please continue with the checkout to complete your order.</strong></p>";
+                                    $('#bread_feedback').html(approved);
                                     $('body').trigger('hideLoadingPopup');
                                 }
+                            }).fail(function(error) {
+                                var errorInfo = {
+                                    bread_config: breadConfig,
+                                    tx_id: tx_token,
```

Comment:
would it be helpful to send back the error itself here?

Reply:
These fail blocks are hit when an unhandled exception is hit in the endpoint causing it to return something other than a 200. In those cases, Magento returns the error message as html for a generic error page, so not really anything too helpful unfortunately. Luckily I don't think these will really be hit much, since we handle the errors on the backend side, and return 200's with the json holding a more useful error message

Reply:
Got it, that works